sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py

```diff
@@ -77,13 +75,7 @@ def __init__(self, base_url: str, credential: AsyncTokenCredential, **kwargs: An
                     "authentication_policy": AppConfigRequestsCredentialsPolicy(credential, base_url, id_credential),
                 }
             )
-        elif hasattr(credential, "get_token"):  # AsyncFakeCredential is not an instance of AsyncTokenCredential
-            kwargs.update(
-                {
-                    "authentication_policy": AsyncBearerTokenCredentialPolicy(credential, *credential_scopes, **kwargs),
```

Comment:
Was this `authentication_policy` call unnecessary?

Reply:
but configuration.py uses BearerTokenCredential instead of AsyncBearerTokenCredential.

Reply:
I just pushed the change for this. I'm thinking that the missing of the pop of credential scopes above was the actual bug. And this was the work around they intended for supporting async given the auto generated code.